saleor/order/utils.py

```diff
@@ -87,16 +87,14 @@ def recalculate_order_discounts(order: Order):
         update_order_discount_for_order(
             order,
             order_discount,
-            force_update=True,
-            # The discount is 0 because the total amount was recaluculated based on
-            # lines price
-            already_applied_discount_amount=Money(Decimal(0), currency=order.currency),
-            save_order=False,
         )
         discount_value = order_discount.value
         discount_type = order_discount.value_type
         amount = order_discount.amount
 
+        if not with_discount_event:
+            continue
```

Comment:
I would say it should be:
```
if with_discount_event:
    # action
```

Reply:
This will add one additional injection to the code :P This will cause that the next condition will be hard to understand but it is up to you